fdbserver/ClusterController.actor.cpp

```diff
@@ -110,17 +111,28 @@ class ClusterControllerData {
 		{
 		}
 
-		void setDistributor(const DataDistributorInterface& distributorInterf) {
+		void setDistributor(const DataDistributorInterface& interf) {
 			ServerDBInfo newInfo = serverInfo->get();
 			newInfo.id = g_random->randomUniqueID();
-			newInfo.distributor = distributorInterf;
+			newInfo.distributor = interf;
 			serverInfo->set( newInfo );
 		}
 
-		void clearDistributor() {
+		void setRatekeeper(const RatekeeperInterface& interf) {
 			ServerDBInfo newInfo = serverInfo->get();
 			newInfo.id = g_random->randomUniqueID();
-			newInfo.distributor = Optional<DataDistributorInterface>();
+			newInfo.ratekeeper = interf;
+			serverInfo->set( newInfo );
```

Comment:
Does there exist a super small chance that the new random UID is the same with the old UID? If this happens, what will be the (bad) effect?

Reply:
I think there is a chance of collision. However, this will happen after a very long period, depending on the pseudo-random algorithm. The underlying code seems to use 32bits random number generator, i.e., [std::mt19937](http://www.cplusplus.com/reference/random/mt19937/), which has an internal 19937 bits state. So the time for the repeat to happen will take roughly O(2^32) calls, which should be a much longer time than the life time of any roles in the cluster. As a result, there is no need to worry about such collisions.